src/lib/gateway-base.ts

```diff
@@ -7,13 +15,30 @@ class GatewayBase {
   private readonly endpoint: string;
   private readonly certs: StarkExCertsConfig;
 
-  constructor(config: StarkExClientConfig, path: string) {
+  constructor(
+    config: StarkExClientConfig,
+    private readonly path: ApiGatewayPath
+  ) {
     const {endpoint, certs} = config;
-    this.initLogger(path);
-    this.endpoint = `${endpoint}${path}`;
-    if (certs) {
-      this.certs = certs;
-    }
+    this.endpoint = endpoint;
+    this.certs = certs;
+
+    this.initLogger(this.getEndpoint());
+  }
+
+  private getEndpoint(options?: {version: ApiVersion}) {
```

Comment:
Why do we pass `options` and not only `version`?
Is this kind of preparation for something?

Reply:
No preparation, it's just a practice I like. It's optional, so don't want to enforce it if we add an other parameter in the future. I agree it's a bit redundant in this case so I'll change it.